dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -304,6 +304,35 @@ def _define_contig_flag(x):
     return flag
 
 
+def _define_dim_flags(x, pos):
+    """
+    Define useful flags for the main calculation in dpnp_matmul.
+    x_is_1D: `x` is 1D array or inhernetly 1D (all dimesnsion are one except for one of them), for instance,
+    if x.shape = (1, 1, 1, 2), then x_is_1D = True
+    x_is_2D: `x` is 2D array or inhernetly 2D (all dimesnsion are one except for the last two of them), for instance,
+    if x.shape = (1, 1, 3, 2), then x_is_2D = True
+    x_base_is_1D: `x` is 1D considering only its last two dimensions, for instance,
+    if x.shape = (3, 4, 1, 2), then x_base_is_1D = True
+    """
+
+    index = -1 if pos == 0 else -2
+    x_shape = x.shape
+    x_ndim = x.ndim
+    x_is_1D = x_ndim == 1
+    if numpy.prod(x_shape) != 0:
+        x_is_1D = x_is_1D or numpy.prod(x_shape) == x_shape[index]
```

Comment:
This `x_is_1D = x_is_1D or numpy.prod(x_shape) == x_shape[index]` returns `True` if `x_is_1D`  without checking the second condition, however, `x_is_1D |= numpy.prod(x_shape) == x_shape[index]` always check the condition on right hand side and it would be a problem for 1D array where the index may be out of range.